src/types.js

```diff
@@ -113,3 +113,15 @@ export function isEnumValue(enumObj, s) {
   }
   return false;
 }
+
+/**
+ * Checks if the given key is a property in the map.
+ *
+ * @param {T}  obj a map like property.
+ * @param {string}  key
+ * @return {boolean}
+ * @template T
+ */
+export function hasOwn(obj, key) {
```

Comment:
`map()` is in this file and I thought this function is closely related to that. Happy to create `src/utils/object.js`.